src/shared_modules/content_manager/src/components/pubSubPublisher.hpp

```diff
@@ -31,17 +61,30 @@ class PubSubPublisher final : public AbstractHandler<std::shared_ptr<UpdaterCont
      *
      * @param context updater context.
      */
-    void publish(const UpdaterContext& context) const
+    void publish(UpdaterContext& context) const
     {
         // If there is data to publish, send it
         if (context.data.contains("paths") && !context.data.at("paths").empty())
         {
             // serialize the JSON object
-            const auto stringifyJson = context.data.dump();
+            const auto message = context.data.dump();
+
+            logDebug2(WM_CONTENTUPDATER, "Data to be published: '%s'", message.c_str());
+
+            std::atomic<bool> {false};
+            const auto [offset, hash, status] = context.spUpdaterBaseContext->fileProcessingCallback(
+                message, context.spUpdaterBaseContext->spStopCondition);
+
+            // If we were processing offsets and it failed, we need to trigger a snapshot to recover
+            if (context.data.at("type") == "offsets" && !status)
```

Comment:
Here you should catch if there was an error in the snapshot processing, currently if there is a failure the chain continues and the offset is set even if the content has not been processed.

![image](https://github.com/user-attachments/assets/c4d3f331-3643-4fe3-8ec2-264acd95f65f)

Reply:
Added a handler for this scenario